Handle long strings in `Select` component

jakubcolony opened the pull request:
## Description

This PR adds a little fix to correctly apply ellipsis overflow to Select options:
<img width="485" alt="Screenshot 2022-10-25 at 19 46 57" src="https://user-images.githubusercontent.com/112586815/197857646-90d38116-e8d7-4a2b-8c26-58dc73ec5b9c.png">

I have checked that it doesn't impact any Select that it shouldn't but would appreciate if you double check that.

Resolves #3991

chinins reviewed (approved):
It seems to be working well.

willm30 reviewed (changes requested):
I think some of the selectors may need to have a min-width added manually where they are used, e.g: 

![no](https://user-images.githubusercontent.com/64402732/197987048-f4cef91c-c971-4b5a-90a0-ebfe23c58e44.png)

Also, should the "selected" label remain visible in its entirety and the team name the thing that is shortened? Just thinking out loud here. An easy fix for selectors that contain domain names (which are capped at 20 chars) would be to remove the right hand side padding, but then you'd need to manually review all other instances and reapply padding where appropriate, so there may be more a general solution. That's if it's worth implementing at all. 

![sel](https://user-images.githubusercontent.com/64402732/197984839-7e683931-e9aa-4817-9159-9393d007d133.png)

willm30 commented:
If the above is seems to much of a hassle, maybe this could be a safe control specific fix? Unless there are any selectors on master that would currently benefit, but I haven't noticed any so far. Domain names are capped at 20 chars and tokens at 5 so they aren't at risk of overflow.

jakubcolony commented:
@willm30 Thanks for spotting this mate, the issue was that I constrained all the Select option lists to 100% width while there were 4 cases in which the list had to be wider, I added an extra appearance prop to handle this:
<img width="225" alt="Screenshot 2022-10-26 at 20 04 42" src="https://user-images.githubusercontent.com/112586815/198114963-c59d0be5-4e3c-4f37-aa94-03c0e9754173.png">
<img width="208" alt="Screenshot 2022-10-26 at 20 04 36" src="https://user-images.githubusercontent.com/112586815/198114972-9bc5624e-a7c1-41bc-812f-72ece11bba3c.png">

>Also, should the "selected" label remain visible in its entirety and the team name the thing that is shortened?

I agree, updated:
<img width="491" alt="Screenshot 2022-10-26 at 19 42 15" src="https://user-images.githubusercontent.com/112586815/198115084-fda8c79e-01df-4ca5-8d49-32b087a9c2ac.png">

I've now manually gone over every (I think) `<Select>` in the dapp and everything seems to be working correctly but if you could have another look that'd be great 👍

willm30 reviewed (approved):
Nice one! From my testing, all is working as expected.
